docs/sdg/sdg-refactor.md

```diff
@@ -0,0 +1,147 @@
+    - 1.1. They have raw seed content.  They want to run the full pipeline including SDG and model training and evaluation.
+    - 1.2. They have raw seed content.  They want to run SDG and then evaluate an existing model on the outputs of that SDG.
+    - 1.3. They have raw seed content.  They want to run SDG only.
+        - 1.3.1. They also want to see the inputs to SDG that get extracted from the raw seed content (i.e., a set of seed context/question/answer tuples and with document chunks for the knowledge if any).
+        - 1.3.2. Alternatively, maybe they only want to see the inputs to SDG -- they don't actually want to run SDG.
+    - 1.4. They have processed seed content.  They want to run the full pipeline including SDG and model training and evaluation.
+    - 1.5. They have processed seed content.  They want to run SDG and then evaluate an existing model on the outputs of that SDG.
+    - 1.6. They have processed seed content.  They want to run SDG only.
+2. User installs the SDG library only.  They want to invoke any of the following as a library call from code they write:
+    - 2.1. They have raw seed content.  They want to run SDG only without any postprocessing.
+    - 2.2. They have processed seed content.  They want to run SDG only without any postprocessing.
+
+If I understand the latest guidance from our product management, the flows that our users want us to support here are 1.1., 1.2., 1.3, 1.3.1, and 1.6.  In an earlier draft of this proposal, I had said that I thought product management also wanted 2.2, but the latest guidance doesn't seem consistent with that understanding.  I am still not sure, so more clarification would be helpful.
+
+## Question 2: What should the commands be in the CLI?
+
+One way to support both 1.3.1 and 1.3.2 would be to have separate CLI commands for the preprocessing, core SDG, and postprocessing step .  Alternatively, a single CLI command that does all of these and also saves the outputs of preprocessing to disk would support 1.3.1 but *not* 1.3.2.  Even if we only want to support 1.3.1, having separate CLI commands for each step might be desirable because it is just more intuitive that if a user wants to save the outputs of preprocessing to disk to have a command to do that instead of having it be a "side effect" of an omnibus SDG command.  Here is a rough outline of what the separate commands would be:
+
+- `ilab data prep` would handle all the preprocessing (the first three bullets in the Context section above, plus any additional preprocessing we add in the future).
+- `ilab data generate` would take as input some data in the same format that `ilab data prep` produces and would run the core synthetic data generation *only*.  Note that this is a breaking change from the current behavior of `ilab data generate`, but that may be acceptable because the user base is still small.
```

Comment:
@cdoern and I discussed this, and he felt strongly that providing backward compatibility was not worth the cost.  In general, the value of backward compatibility is high when the user base is large relative to the expected future user base.  I think that's mostly not the case for InstructLab, so I agree with @cdoern .

Reply:
FWIW, I have updated the document to leave both options open since we don't seem to be resolving this here, and we want the rest of what's in this document to merge if possible.